src/utils/smartWallet.js

```diff
@@ -91,6 +91,16 @@ export const userHasSmartWallet = (accounts: Accounts = []): boolean => {
   return accounts.some(acc => acc.type === ACCOUNT_TYPES.SMART_WALLET);
 };
 
+export const getPreferredWalletId = (accounts: Accounts = []): string => {
```

Comment:
you can just use the `getActiveAccountWalletId()` function instead

Reply:
@jsidorenko not sure about this, looking at whole implementation this seems that this method and whole logic describes that this returns "default wallet address to receive exchange result" and based on PR we see that we can either receive to our Smart Wallet if it's deployed and otherwise – Key Based. But then there's no switch as per designs so not sure if the logic in this PR follows the design @alazarevski?

Reply:
Per discussion with @paragramm and @vitorpy, we will use the SmartWallet walletId as the default receiver, even though the active account might be legacy. The only exception in this method is if there is no smart-wallet at all (edge case) is when we return the legacy walletId.